cephfs/admin/response.go

```diff
@@ -0,0 +1,101 @@
+// +build !luminous,!mimic
+
+package admin
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+)
+
+var (
+	// ErrStatusNotEmpty may be returned if a call should not have a status
+	// string set but one is.
+	ErrStatusNotEmpty = errors.New("response status not empty")
+	// ErrBodyNotEmpty may be returned if a call should have an empty body but
+	// a body value is present.
+	ErrBodyNotEmpty = errors.New("responase body not empty")
+)
+
+// response encapsulates the data returned by ceph and supports easy processing
+// pipelines.
+type response struct {
+	body   []byte
+	status string
+	err    error
+}
+
+// Ok returns true if the response contains no error.
+func (r response) Ok() bool {
```

Comment:
Is there a reason why the functions here do not take a pointer to the `response`?

Reply:
The functions are not meant to mutate r and thus either return a new response or something else (in this example the bool). There's also the fact that I recently read that most go code over uses pointers. I don't know if this is [the article I read](https://medium.com/@meeusdylan/when-to-use-pointers-in-go-44c15fe04eac) but it touches on the issues I recall. Long story short: I didn't need to be a pointer so I didn't make it one. :-)

Reply:
Ok, if there are good reasons :+1: